components/telemetry-operator/internal/fluentbit/emitter.go

```diff
@@ -0,0 +1,85 @@
+package fluentbit
+
+import (
+	"fmt"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+type SectionBuilder struct {
+	indentation string
+	valueTab    string
+	builder     strings.Builder
+}
+
+func NewSectionBuilder() *SectionBuilder {
+	return &SectionBuilder{
+		indentation: strings.Repeat(" ", 4),
+		valueTab:    strings.Repeat(" ", 22),
+	}
+}
+
+func (sb *SectionBuilder) CreateFilterSection() *SectionBuilder {
+	sb.builder.WriteString("[FILTER]")
+	sb.builder.WriteByte('\n')
+	return sb
+}
+
+func (sb *SectionBuilder) AddConfigurationParameter(key string, value string) *SectionBuilder {
+	sb.builder.WriteString(fmt.Sprintf("%s%s%s%s",
+		sb.indentation,
+		key,
+		sb.valueTab[:len(sb.valueTab)-len(key)],
+		value))
+	sb.builder.WriteByte('\n')
+	return sb
+}
+
+func (sb *SectionBuilder) ToString() string {
+	sb.builder.WriteByte('\n')
+	return sb.builder.String()
+}
+
+// CreateEmitter creates the Fluent Bit Rewrite Tag Filter section
+func CreateEmitter(config PipelineConfig, logPipeline *telemetryv1alpha1.LogPipeline) string {
```

Comment:
Not sure about the naming here. I like the idea of extracting this piece of logic into a separate file, but I'd rather call it `rewrite_tag` / `CreateRewriteTagSection` or something. Otherwise, it can be confused with rewrite tag's emitter fields: https://docs.fluentbit.io/manual/pipeline/filters/rewrite-tag.